packages/tpetra/core/test/LinearProblem/LinearProblem_UnitTests.cpp

```diff
@@ -0,0 +1,305 @@
+    using LPT = Tpetra::LinearProblem<Scalar,LO,GO,Node>;
+    //using local_ordinal_type = typename Tpetra::Vector<Scalar, LO, GO, Node>::local_ordinal_type;
+    using global_ordinal_type = typename Tpetra::Vector<Scalar, LO, GO, Node>::global_ordinal_type;
+
+    
+    const global_ordinal_type INVALID = Teuchos::OrdinalTraits<global_ordinal_type>::invalid();
+    constexpr bool debug = true;
+
+    RCP<Teuchos::FancyOStream> outPtr = debug ?
+      Teuchos::getFancyOStream (Teuchos::rcpFromRef (std::cerr)) :
+      Teuchos::rcpFromRef (out);
+    Teuchos::FancyOStream& myOut = *outPtr;
+    Teuchos::OSTab tab0 (myOut);
+
+    myOut << "Test: LinearProblem, Constructors" << endl;
+
+    RCP<const Comm<int> > comm = getDefaultComm();
+    //const size_t numImages = comm->getSize();
+    const size_t myImageID = comm->getRank();
+    // create a Map
+    const size_t numLocal = 5;
+    const size_t numVecs  = 1;
+    RCP<const map_type> map = createContigMapWithNode<LO,GO,Node>(INVALID,numLocal,comm);
+    GO base = numLocal*myImageID;
+    GST globalNumElements = map->getGlobalNumElements();
+    RCP<Tpetra::RowMatrix<Scalar,LO,GO,Node> > A;
+    {
+      RCP<MAT> A_crs = rcp(new MAT(map,3));
+      for (size_t i=0; i<numLocal; ++i) {
+        A_crs->insertGlobalValues(base + i, tuple<GO>(base + i), tuple<Scalar>(2.0)); // Diagonal entry
+
+        GST globalIndex = base + i;
+        // Insert the first subdiagonal entry if not in the first row
+        if (globalIndex > 0) {
+            A_crs->insertGlobalValues(base + i, tuple<GO>(base + i - 1), tuple<Scalar>(1.0)); // Subdiagonal entry
+        }
+
+        // Insert the first superdiagonal entry if not in the last row
+        if (globalIndex < globalNumElements - 1) {
+            A_crs->insertGlobalValues(base + i, tuple<GO>(base + i + 1), tuple<Scalar>(1.0)); // Superdiagonal entry
+        }
+      }
+      A_crs->fillComplete();
+      A = A_crs;
+    }
+
+    // create solution, rhs and scaling vector
+    RCP<MV> X = rcp (new MV (map, numVecs));
+    RCP<MV> B = rcp (new MV (map, numVecs));
+    RCP<VT> S = rcp (new VT (map));
+
+    // Assign values to the MultiVector based on the global index
+    for (size_t localIndex = 0; localIndex < numLocal; ++localIndex) {
+        auto globalIndex = map->getGlobalElement(localIndex);
+        S->replaceLocalValue(localIndex, Teuchos::as<Scalar>(globalIndex + 1));
+        for (size_t j = 0; j < numVecs; ++j) { // Loop over each vector (column)
+                // Assign a value (for example, the global index plus the vector index)
+                X->replaceLocalValue(localIndex, j, Teuchos::as<Scalar>(globalIndex + j + 1));
+                B->replaceLocalValue(localIndex, j, Teuchos::as<Scalar>(globalIndex + j + 1));
+        }
+    }
+
+    RCP<LPT> linearProblem = rcp(new LPT());
+
+    linearProblem->setMatrix(A);
+    linearProblem->setLHS(X);
+    linearProblem->setRHS(B);
+
+    linearProblem->checkInput();
```

Comment:
`linearProblem->checkInput()` could be called before setting A, X, and B, respectively, to test the error checking.